SemiLeptonicWVA/Utils/src/GenEventInfo.cc

```diff
@@ -264,6 +272,23 @@ GenEventInfo::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	  PUWeight = LumiWeights_.weight( Tnpv );
 	}
 
+
+
+	edm::Handle< LHEEventProduct > LHEEventInfo;
+        iEvent.getByLabel("source", LHEEventInfo);
+
+	std::vector<double> AQGCweights;
+        double originalWeight;
+
+	if (LHEEventInfo.isValid()){
+	  auto weightsTemp = LHEEventInfo ->weights();
+	  originalWeight *= LHEEventInfo ->originalXWGTUP();
```

Comment:
Remove the asterisk and just change to:  originalWeight = LHEEventInfo ->originalXWGTUP();